ccmlib/scylla_repository.py

```diff
@@ -244,7 +245,15 @@ def setup(version, verbose=True, skip_downloads=False):
         type_n_version = version.split(os.path.sep, 1)
     version_dir = version_directory(version) if not skip_downloads else None
 
-    if len(type_n_version) == 2 and version_dir is None:
+    # If the test version is unstable (not release, maybe private branch) and installation folder exists,
+    # we want to check if this version was downloaded nd installed already in the past and was changed.
+    # In this case the version should be downloaded and installed again.
+    # Compare hash of saved version (it is saved in the 'scylla-core-package/source.txt' file) and hash of a new package.
+    # If it is same - skip the download. If not - remove existing folder and download again.
+    # TODO: is it good condition?
+    validate_by_hash = version_dir is not None and type_n_version[0] != "release"
```

Comment:
@fruch  is ti good enough condition?

Reply:
I would save one call for the release case

Reply:
so yes let's keep it